streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -838,6 +838,21 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
         if (firstException.get() != null) {
             throw firstException.get();
         }
+
+        removeRevokedTasksFromStateUpdater(remainingRevokedPartitions);
+    }
+
+    private void removeRevokedTasksFromStateUpdater(final Set<TopicPartition> remainingRevokedPartitions) {
+        if (stateUpdater != null) {
+            for (final Task restoringTask : stateUpdater.getTasks()) {
+                if (restoringTask.isActive()) {
+                    if (remainingRevokedPartitions.containsAll(restoringTask.inputPartitions())) {
```

Comment:
Should we also remove those partitions from the `remainingRevokedPartitions` as well?

Reply:
Ack